internal/db/db_sqlite_test.go

```diff
@@ -514,3 +514,165 @@ func (s *SqliteSuite) TestFindUser_DoesNotExist() {
+			Size:       11,
+			RawContent: []byte("hello world"),
+			Private:    false,
+			Type:       "plaintext",
+			UserID:     id.New(),
+		},
+		{
+			ID:         id.New(),
+			CreatedAt:  time.Now().UTC(),
+			UpdatedAt:  time.Now().UTC(),
+			Size:       11,
+			RawContent: []byte("hello world, again"),
+			Private:    false,
+			Type:       "plaintext",
+			UserID:     id.New(),
+		},
+	}
+
+	const query = `
+		INSERT INTO files (
+			id,
+			created_at,
+			updated_at,
+			size,
+			content,
+			private,
+			type,
+			user_id
+		) VALUES (?, ?, ?, ?, ?, ?, ?, ?)`
+
+	for _, existingFile := range existingFiles {
+		_, err := s.testDB.Exec(
+			query,
+			existingFile.ID,
+			existingFile.CreatedAt,
+			existingFile.UpdatedAt,
+			existingFile.Size,
+			existingFile.RawContent,
+			existingFile.Private,
+			existingFile.Type,
+			existingFile.UserID,
+		)
+		s.Require().NoError(err)
+	}
+
+	filesPageOne, err := database.LatestPublicFiles(context.TODO(), 1, 1)
+
+	// for some reason the raw content returns nil
+	expected := existingFiles[1]
+	expected.RawContent = nil
+
+	s.Require().NoError(err)
+	s.Require().Equal(expected, filesPageOne[0])
+
+	filesPageTwo, err := database.LatestPublicFiles(context.TODO(), 2, 1)
+
+	// for some reason the raw content returns nil
+	expected = existingFiles[0]
+	expected.RawContent = nil
+
+	s.Require().NoError(err)
+	s.Require().Equal(existingFiles[0], filesPageTwo[0])
+}
+
+func (s *SqliteSuite) TestLatestPublicFiles_PublicOnly() {
+	database := s.getTestDB(true)
+
+	existingFiles := []*snips.File{
+		{
+			ID:         id.New(),
+			CreatedAt:  time.Now().UTC(),
+			UpdatedAt:  time.Now().UTC(),
+			Size:       11,
+			RawContent: []byte("hello world"),
+			Private:    false,
+			Type:       "plaintext",
+			UserID:     id.New(),
+		},
+		{
+			ID:         id.New(),
+			CreatedAt:  time.Now().UTC(),
+			UpdatedAt:  time.Now().UTC(),
+			Size:       11,
+			RawContent: []byte("hello world, I am private"),
+			Private:    true,
+			Type:       "plaintext",
+			UserID:     id.New(),
+		},
+		{
+			ID:         id.New(),
+			CreatedAt:  time.Now().UTC(),
+			UpdatedAt:  time.Now().UTC(),
+			Size:       11,
+			RawContent: []byte("hello world, again"),
+			Private:    false,
+			Type:       "plaintext",
+			UserID:     id.New(),
+		},
+	}
+
+	const query = `
+		INSERT INTO files (
+			id,
+			created_at,
+			updated_at,
+			size,
+			content,
+			private,
+			type,
+			user_id
+		) VALUES (?, ?, ?, ?, ?, ?, ?, ?)`
+
+	for _, existingFile := range existingFiles {
+		_, err := s.testDB.Exec(
+			query,
+			existingFile.ID,
+			existingFile.CreatedAt,
+			existingFile.UpdatedAt,
+			existingFile.Size,
+			existingFile.RawContent,
+			existingFile.Private,
+			existingFile.Type,
+			existingFile.UserID,
+		)
+		s.Require().NoError(err)
+	}
+
+	filesPageOne, err := database.LatestPublicFiles(context.TODO(), 1, 1)
+
+	// for some reason the raw content returns nil
+	expected := existingFiles[2]
```

Comment:
This is expected, since `LatestPublicFiles` does not have raw content being sent in the query

Reply:
You're right, potato me 😅

Reply:
🥔